transport/gdrive-transport/src/main/java/org/apache/airavata/mft/transport/gdrive/GDriveMetadataCollector.java

```diff
@@ -0,0 +1,146 @@
+        GDriveSecret gdriveSecret = secretClient.getGDriveSecret(GDriveSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
+
+
+        HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = new JacksonFactory();
+        String jsonString = gdriveSecret.getCredentialsJson();
+        GoogleCredential credential = GoogleCredential.fromStream(new ByteArrayInputStream(jsonString.getBytes(StandardCharsets.UTF_8)), transport, jsonFactory);
+        if (credential.createScopedRequired()) {
+            Collection<String> scopes = DriveScopes.all();
+            credential = credential.createScoped(scopes);
+        }
+
+        Drive drive = new Drive.Builder(transport, jsonFactory, credential).build();
+        ResourceMetadata metadata = new ResourceMetadata();
+        FileList fileList = drive.files().list()
+                .setQ("name = '"+gdriveResource.getResourcePath()+"'")
+                .setFields("files(id,name,modifiedTime,md5Checksum,size,mimeType)")
+                .execute();
+
+        for (File f : fileList.getFiles()) {
+                metadata.setMd5sum(f.getMd5Checksum());
+                metadata.setUpdateTime(f.getModifiedTime().getValue());
+                metadata.setResourceSize(f.getSize().longValue());
+        }
+
+        return metadata;
+    }
+
+    @Override
+    public Boolean isAvailable(String resourceId, String credentialToken) throws Exception {
+        checkInitialized();
+        String id = null;
+        ResourceServiceGrpc.ResourceServiceBlockingStub resourceClient = ResourceServiceClient.buildClient(resourceServiceHost, resourceServicePort);
+        GDriveResource gdriveResource = resourceClient.getGDriveResource(GDriveResourceGetRequest.newBuilder().setResourceId(resourceId).build());
+
+        SecretServiceGrpc.SecretServiceBlockingStub secretClient = SecretServiceClient.buildClient(secretServiceHost, secretServicePort);
+        GDriveSecret gdriveSecret = secretClient.getGDriveSecret(GDriveSecretGetRequest.newBuilder().setSecretId(credentialToken).build());
+
+
+        HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = new JacksonFactory();
+        String jsonString = gdriveSecret.getCredentialsJson();
+        GoogleCredential credential = GoogleCredential.fromStream(new ByteArrayInputStream(jsonString.getBytes(StandardCharsets.UTF_8)), transport, jsonFactory);
+        if (credential.createScopedRequired()) {
+            Collection<String> scopes = DriveScopes.all();
+            credential = credential.createScoped(scopes);
+        }
+
+        Drive drive = new Drive.Builder(transport, jsonFactory, credential).build();
+
+        FileList fileList = drive.files().list()
+                .setQ("name = '"+gdriveResource.getResourcePath()+"'")
+                .setFields("files(id,name)")
+                .execute();
+
+        for (File f : fileList.getFiles()) {
```

Comment:
Validate for fileList.getFiles() empty and having more than 1 entry

Reply:
We have handled for empty case. For loop should work if there are two files with same names.